addon/components/au-time-picker.js

```diff
@@ -1,16 +1,52 @@
 import Component from '@glimmer/component';
 import { action } from '@ember/object';
-import { tracked } from '@glimmer/tracking';
 import { isPresent } from '@ember/utils';
+import { trackedReset } from 'tracked-toolbox';
+import { formatTimeDigit } from '../helpers/format-time-digit';
 
 export default class AuTimePickerComponent extends Component {
-  @tracked hourValue = this.args.hours || 12;
-  @tracked minuteValue = this.args.minutes || 0;
-  @tracked secondValue =
-    isPresent(this.args.showSeconds) && !this.args.showSeconds
-      ? 0
-      : this.args.seconds || 0; // ignore 'seconds' argument if 'showSeconds' is false
-  @tracked keyCodes = [8, 9, 13, 33, 34, 37, 39, 46];
+  @trackedReset({
+    memo() {
+      return this.validateTimeValue(this.args.hours, 'hourValue');
+    },
+    update() {
+      return this.validateTimeValue(this.args.hours, 'hourValue');
+    },
+  })
+  hourValue = 12;
```

Comment:
It seems the default value is no longer used at the moment? It seems to default to 0. Does that mean that the update function also runs for the initial value?

Reply:
I've been trying to figure this out too, but it seems like it. It works in a weird way. The documentation could be much better for this.
If the memo is a function, then the default value `= 12;` is just ignored. But when memo is a path `args.hours` then the default `= 12;` takes effect. The update function will run on initial construction, but not when the default value is taken.